### PR TITLE
KAFKA-15593: Add 3.6 to core upgrade and compatibility tests

### PR DESCRIPTION
**Pull request by satishd**
This is part of the 3.6.0 release process as mentioned [here](https://cwiki.apache.org/confluence/display/KAFKA/Release+Process).

```
For feature releases, file a JIRA for updating compatibility/upgrade system tests to test the newly released version. 
```

This PR is only related to core upgrade and compatibility related tests.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by clolov — APPROVED** [review]
This looks good to me, thank you!

File: tests/kafkatest/tests/core/compatibility_test_new_broker_test.py [hunk 1]
```diff
@@ -21,9 +21,10 @@
 from kafkatest.services.zookeeper import ZookeeperService
 from kafkatest.tests.produce_consume_validate import ProduceConsumeValidateTest
 from kafkatest.utils import is_int
-from kafkatest.version import LATEST_0_8_2, LATEST_0_9, LATEST_0_10_0, LATEST_0_10_1, LATEST_0_10_2, LATEST_0_11_0, \
+from kafkatest.version import (LATEST_0_8_2, LATEST_0_9, LATEST_0_10_0, LATEST_0_10_1, LATEST_0_10_2, LATEST_0_11_0, \
```
Comment (jlprat) [inline tests/kafkatest/tests/core/compatibility_test_new_broker_test.py]: Why the new parens, and only in this file and not in the others?

**Review by jlprat — APPROVED** [review]
Thanks @satishd LGTM
I just have a nitpick about parenthesis on import only being present on one of the files

**Comment by jlprat** [comment]
I restarted the jenkins job as lots of tests failed at the initialization step

**Comment by mjsax** [comment]
Seems this does not cover KS -- btw: there is some issue for KS system tests right now, and as a matter of fact, we did not even add the missing upgrade tests in older release yet (PR is still pending).

I am ok with merging this PR, but we should not close the JIRA until everything is resolved IMHO.

**Comment by satishd** [comment]
>Seems this does not cover KS -- btw: there is some issue for KS system tests right now, and as a matter of fact, we did not even add the missing upgrade tests in older release yet (PR is still pending).

Right, this change is related to core upgrade and compatibility tests. Updated the PR description for clarity. I saw that KS related updates for earlier releases are in progress as part of https://github.com/apache/kafka/pull/13860

>I am ok with merging this PR, but we should not close the JIRA until everything is resolved IMHO.

Sure, created two separate sub tasks on [KAFKA-15576](https://issues.apache.org/jira/browse/KAFKA-15576) for tracking them separately and close the parent JIRA once both the sub tasks are resolved.

**Pull request merged**